lib/ansible/modules/cloud/amazon/iam_cert_facts.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/python
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: iam_cert_facts
+short_description: Retrieves iam_cert details using AWS methods.
+description:
+    - Gets various details related to IAM cert, ARN, expiration, Certificates.
+version_added: "2.4"
+options:
+  certificate_name:
+    description:
+      - The name of the server certificate you want to retrieve information about.
+    required: False
+
+  path_prefix:
+    description:
+      - The path prefix for filtering the results. http://docs.aws.amazon.com/cli/latest/reference/iam/list-server-certificates.html#options
+    required: False
+
+
+author: Julien Thebault (@Lujeni)
+extends_documentation_fragment: aws
+'''
+
+EXAMPLES = '''
+- name: List all server certificates
+  iam_cert_facts:
+  register: server_certificates
+
+- name: Retrieve specific certificate
+  iam_cert_facts:
+      certificate_name: cookie
+
+- name: List all server certificates with this path
+  iam_cert_facts:
+      path_prefix: /company/servercerts
+'''
+
+RETURN = '''
+is_truncated:
+    description: A flag that indicates whether there are more items to return.
+    returned: always
+    type: bool
+    sample: false
+
+response_metadata:
+    description: Response metadata.
+    returned: always
+    type: dict
+    sample: {"http_status_code": 200, "request_id": "9f6d-671298e72d8b"}
+
+server_certificate_metadata_list:
+    description: Metadata list.
+    returned: always
+    type: dict
+    sample: http://boto3.readthedocs.io/en/latest/reference/services/iam.html#IAM.Client.list_server_certificates
+'''
+
+try:
+    import boto3
+    import botocore
```

Comment:
botocore isn't used anywhere and you're import ClientError below so I think you can remove this